pkg/infinispan/configuration/server/configuration_test.go

```diff
@@ -0,0 +1,53 @@
+package server
+
+import (
+	"os"
+	"testing"
+
+	"github.com/blang/semver"
+	"github.com/infinispan/infinispan-operator/pkg/infinispan/version"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGenerateAndGenerateZeroCapacity(t *testing.T) {
+	baseCfg_expected := ReadFile("test/base-15-cfg.xml")
+	adminCfg_expected := ReadFile("test/admin-15-cfg.xml")
+	zeroCfg_expected := ReadFile("test/zero-15-cfg.xml")
+
+	ispn := Infinispan{Authorization: &Authorization{Enabled: true}}
+	spec := Spec{Infinispan: ispn}
+	vers := semver.Version{Major: 15, Minor: 1, Patch: 25}
+	ope := version.Operand{UpstreamVersion: &vers}
+
+	baseCfg, adminCfg, err := Generate(ope, &spec)
+	assert.Equal(t, baseCfg, baseCfg_expected)
+	assert.Equal(t, adminCfg, adminCfg_expected)
+	assert.Nil(t, err)
+
+	zeroCfg, err := GenerateZeroCapacity(ope, &spec)
+	assert.Equal(t, zeroCfg, zeroCfg_expected)
+	assert.Nil(t, err)
+}
+
+func TestGenerate_v14(t *testing.T) {
+	baseCfg_expected := ReadFile("test/base-14-cfg.xml")
+	adminCfg_expected := ReadFile("test/admin-14-cfg.xml")
+
+	ispn := Infinispan{Authorization: &Authorization{Enabled: true}}
+	spec := Spec{Infinispan: ispn}
+	vers := semver.Version{Major: 14, Minor: 0, Patch: 11}
+	ope := version.Operand{UpstreamVersion: &vers}
+
+	baseCfg, adminCfg, err := Generate(ope, &spec)
+	assert.Equal(t, baseCfg, baseCfg_expected)
```

Comment:
In the various `assert.Equal` calls we have the expected and actual arguments in the wrong order, it should be `Equal(t, expected, actual)`.